backend/modules/simulation/agents/grading_agent.py

```
@@ -297,14 +521,142 @@ async def grade_scene(
+                    }
+                    for c in result.criteria_breakdown
+                ],
+                "business_thinking_quality": result.business_thinking_quality,
+                "key_strengths": result.key_strengths,
+                "areas_for_improvement": result.areas_for_improvement,
+                "actionable_recommendations": result.actionable_recommendations,
+                "automated_check_results": auto_results,
+                "scene_type": "code_challenge",
+            }
+
+        except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)